src/elements/Tree.tsx

```diff
@@ -0,0 +1,321 @@
+    handleMouseEnter: (event: React.MouseEvent, node: TreeElementNode) => void,
+    handleMouseLeave: (event: React.MouseEvent, node: TreeElementNode) => void,
+    handleItemClick: (node: any) => void,
+    handleFolderClick: (node: any) => void,
+    handleSecondaryActionClick: (event: React.MouseEvent, node: TreeElementNode) => void
+) {
+    const listItems: JSX.Element[] = [];
+
+    nodes.forEach((node, index) => {
+        listItems.push(
+            renderChild(
+                node,
+                index,
+                value,
+                showCheckBoxes,
+                level,
+                brandingService,
+                handleMouseEnter,
+                handleMouseLeave,
+                handleItemClick,
+                handleFolderClick,
+                handleSecondaryActionClick
+            )
+        );
+    });
+
+    return listItems;
+}
+
+/**
+ * A tree view form element.
+ * @displayName Tree
+ * @description A tree view form element.
+ * @param props The props that will be provided by the Workflow runtime.
+ */
+function TreeElement(props: TreeElementProps): React.ReactElement {
+    const {
+        dense,
+        maxHeight,
+        maxWidth,
+        subheader,
+        value,
+        showCheckBoxes,
+        onClick,
+        onMouseEnter,
+        onMouseLeave,
+        raiseEvent,
+        setValue,
+    } = props;
+    const [items, setItems] = React.useState(props.items);
+
+    const { brandingService } = React.useContext(UIContext);
+
+    const handleMouseEnter = (event: React.MouseEvent, node: TreeElementNode) => {
+        raiseEvent("custom", { eventType: "mouseEnter", node });
+        onMouseEnter?.(node);
+    };
+
+    const handleMouseLeave = (event: React.MouseEvent, node: TreeElementNode) => {
+        raiseEvent("custom", { eventType: "mouseLeave", node });
+        onMouseLeave?.(node);
+    };
+
+    const handleSecondaryActionClick = (event: React.MouseEvent, node: any) => {
+        raiseEvent("custom", { eventType: "secondaryAction", node });
+    };
+
+    const handleFolderClick = (node: TreeElementNode) => {
+        if (node.children && node.children.length > 0) {
+            node.open = !node.open;
+            setItems([...items]);
+        }
+    };
+
+    const handleItemClick = (node: TreeElementNode) => {
+        if (showCheckBoxes) {
+            node.checked = !node.checked;
+            setValue(filterTree(items));
+        } else {
+            setValue([node]);
+        }
+
+        raiseEvent("clicked" as any, node);
+        onClick?.(node);
+    };
+
+    return (
+        <Box
+            maxHeight={maxHeight}
+            maxWidth={maxWidth}
+            sx={{
+                overflowY: "auto",
+                // Hack required to override .gcx-forms.defaults
+                ["& button"]: {
+                    marginRight: "0 !important",
+                },
+            }}
+        >
+            <List dense={dense} subheader={subheader}>
```

Comment:
I'm not sure what the cause is, but it does not render the first time the Tree is used. You have to run it twice.

Reply:
Yea. I'm confused by that too.

Reply:
It has something to do with the props being passed in on the load of the workflow.  I resolved it by using a useEffect hook.

Reply:
Fixed.